### DISCUSSION TITLE
Several conditions for entity icon color

### DISCUSSION THREAD

**Discussion by rom169** [marcokreeft87/room-card#284] [Q&A]
Hi,
trying several conditions to change icon color of entities.

Single condition works:
type: custom:room-card
title: Test
entity: light.lightsstairsegug_2
show_icon: true
info_entities:
  - entity: binary_sensor.sensordoormain_window_door_is_open
    show_icon: true
    icon:
      template:
        styles: |
          if (entity.state == 'on') return 'color: mediumseagreen';

Any hint on how to make something like this work:
type: custom:room-card
title: Test
entity: light.lightsstairsegug_2
show_icon: true
info_entities:
  - entity: binary_sensor.sensordoormain_window_door_is_open
    show_icon: true
    icon:
      template:
        styles: >
          if ((entity.state == 'on') and (states('binary_sensor.anybody_home') == 'on')) return 'color: mediumseagreen';
          if ((entity.state == 'on') and (states('binary_sensor.anybody_home') == 'off')) return 'color: cyan';

Thanks

**Comment by marcokreeft87** [comment]
Try else if as second condition or just use conditions like in the wiki

**Reply by rom169** [reply]
What I want to realize is not an else, I need an AND of two states, one of the entity I'm using in the card and one of another entity:
if ((entity.state == 'on') and (states('binary_sensor.anybody_home') == 'on')) return 'color: mediumseagreen';

I could do that by creating an additional sensor, but although it is covered by the documentation, using icon templating states results in a card syntax error:
if (states('binary_sensor.anybody_home') == 'on') return 'color: mediumseagreen';

**Reply by marcokreeft87** [reply]
Try using && instead of 'and'

**Comment by rom169** [comment]
Did some more testing. This syntax is working:
if ((entity.state == 'on') & (states['binary_sensor.anybody_home'].state == 'on' )) return 'color: mediumseagreen';

**Reply by marcokreeft87** [reply]
I'd change it to && so double &. A single & does bitwise and.
